lib/phases/phase.js

```diff
@@ -111,35 +113,55 @@ Phase.prototype.registerParser = function(parser) {
  * @returns {Promise}
  */
 Phase.prototype.getDevice = function() {
-  var phase = this;
-
   if (this.device) {
     return Promise.resolve(this.device);
   }
 
-  return mozDevice(this.options.serial, this.options)
-    .then(function(device) {
-      phase.device = device;
+  service(this.options)
+    .then(device => this.device = device)
+    .then(() => this.setDeviceMetadata())
+    .then(() => {
+      let device = this.device;
+      let options = this.options;
+      let port = options.forwardPort || options.marionettePort;
 
-      // Allow specific phases to handle their own Dispatcher set up
-      if (phase.options.preventDispatching) {
+      if (options.preventDispatching) {
         return device;
       }
 
-      phase.dispatcher = new Dispatcher(device);
-      device.log.start();
-      phase.registerParser(performanceParser);
-      phase.registerParser(memoryParser);
-
-      return device
-        .adbForward(phase.options.forwardPort || phase.options.marionettePort)
-        .then(function() {
-          return device;
-        });
+      return this
+        .logstream()
+        .then(stream => this.dispatcher = new Dispatcher(stream))
+        .then(() => this.registerParser(performanceParser))
+        .then(() => this.registerParser(memoryParser))
+        .then(() => device.adbForward(port))
+        .then(port => this.marionette = new Marionette(R.merge(options, { port })))
+        .then(() => device);
     })
-    .catch(function(err) {
-      phase.emit('error', err);
-    });
+    .catch(err => this.emit('error', err));
+};
+
+Phase.prototype.logstream = function() {
+  let logcat = this.options.logcat;
+
+  return this.device
+    .logstream()
+    .then(R.apply((stream, request) => {
+      this.streamRequest = request;
+      logcat && this.device.filestream(logcat);
+      return stream;
+    }));
+};
+
+Phase.prototype.setDeviceMetadata = function() {
```

Comment:
No, since its return value is not really consumed and `this.properties` is set from here.